src/picomsg/src/routing_table/c_api/routing_table.h

```diff
@@ -0,0 +1,53 @@
+
+#pragma once
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <stdint.h>
+#include "common_c_api/common.h"
+
+
+// NOTE: These should be identical to 'struct sp_database_sp_info' but without the direction!
+struct routing_table_route_entry_sp_info {
+	struct pm_ipsec_ts traffic_selector;
+	uint16_t vlan_id;
+};
+
+struct routing_table_route_entry {
+	struct routing_table_route_entry_sp_info sp_info;
+
+	// NOTE: Should contain 'struct sl_routes_conn_add_route_params'
+	struct pm_proto_prefix prefix;
+	struct pm_proto_addr nexthop;
+	uint32_t vrf_id;
+
+	bool is_installed;
+};
+
+// Opaque type
+struct routing_table {
+};
+
+struct routing_table* routing_table_create();
+void routing_table_destroy(struct routing_table* table);
+
+bool routing_table_add_route(struct routing_table* table,
+							 const struct routing_table_route_entry* route_entry);
+
+bool routing_table_del_route_by_sp(struct routing_table* table,
+								   const struct routing_table_route_entry_sp_info* route_entry_sp_info,
+								   bool* is_route_installed);
+
+/**
+ * NOTE: @callback is only allowed to modify @is_installed!
+ */
+bool routing_table_call_route_by_sp(struct routing_table* table,
+									const struct routing_table_route_entry_sp_info* route_entry_sp_info,
+									void* ctx,
+									bool (*callback)(void* ctx, struct routing_table_route_entry*));
+
```

Comment:
Weird formatting or maybe it's just Github UI that shows strange alignment of arg types.

Reply:
Github UI butchers it